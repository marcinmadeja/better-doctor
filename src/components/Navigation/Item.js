// @flow strict
import * as React from 'react';
import classnames from 'classnames';
import { NavLink } from 'react-router-dom';
import styled from 'styled-components';
import MenuItem from '@material-ui/core/MenuItem';
import ListItemIcon from '@material-ui/core/ListItemIcon';

const StyledIcon = styled.span`
  display: flex;
  align-items: center;
  height: 100%;
`;

type Props = {
  icon: React.Node,
  children: React.Node,
  path: string,
  className?: string,
};

const Item = ({ icon, path, children, className }: Props) => {
  const c_className: string = className || '';
  const componentClasses = classnames({ [c_className]: className });

  return (
    <MenuItem
      to={path}
      exact
      component={NavLink}
      className={componentClasses}
    >
      <ListItemIcon>
        <StyledIcon>
          {icon}
        </StyledIcon>
      </ListItemIcon>

      {children}
    </MenuItem>
  );
};

Item.defaultProps = {
  className: '',
};

export default Item;
