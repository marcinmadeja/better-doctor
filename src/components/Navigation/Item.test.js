import React from 'react';
import Item from './Item';

const defaultProps = {
  Icon: () => <p />,
  children: '',
};

const setup = (props = {}) => {
  props = { ...defaultProps, ...props };
  const actions = {};
  const component = <Item {...actions} {...props} />;
  const shallowComponent = shallow(component);

  return {
    actions,
    component,
    shallowComponent,
  };
};

describe('Item', () => {
  it('should render', () => {
    const { shallowComponent } = setup();
    expect(shallowComponent.exists()).toBe(true);
  });
});
