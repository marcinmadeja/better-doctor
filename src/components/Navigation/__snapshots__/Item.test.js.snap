// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Item should match base snapshot 1`] = `
<a
  aria-current={null}
  className="MuiButtonBase-root-13 MuiListItem-root-3 MuiListItem-default-6 MuiListItem-gutters-10 MuiListItem-button-11 MuiMenuItem-root-1 example class name"
  href="/example/path"
  onBlur={[Function]}
  onClick={[Function]}
  onFocus={[Function]}
  onKeyDown={[Function]}
  onKeyUp={[Function]}
  onMouseDown={[Function]}
  onMouseLeave={[Function]}
  onMouseUp={[Function]}
  onTouchEnd={[Function]}
  onTouchMove={[Function]}
  onTouchStart={[Function]}
  role="menuitem"
  style={undefined}
  tabIndex={-1}
>
  <span
    className="MuiListItemIcon-root-16 sc-bdVaJa gguFeu"
  >
    <p />
  </span>
  Lorem ipsum
  <span
    className="MuiTouchRipple-root-17"
  />
</a>
`;
